File: android/src/main/java/dev/barooni/capacitor/calendar/models/inputs/OpenCalendarInput.kt
package dev.barooni.capacitor.calendar.models.inputs

import com.getcapacitor.PluginCall
import dev.barooni.capacitor.calendar.utils.ImplementationHelper

data class OpenCalendarInput(
    private val call: PluginCall,
) {
    val date: Long = ImplementationHelper.getCalendarFromTimestamp(call.getLong("date")).timeInMillis
}

File: ios/Plugin/Models/Inputs/OpenCalendarInput.swift
import Capacitor

struct OpenCalendarInput {
    private var date: Double?

    init(call: CAPPluginCall) {
        self.date = call.getDouble("date")
    }

    func getDate() -> TimeInterval {
        if let date = date {
            return ImplementationHelper.dateFromTimestamp(date).timeIntervalSinceReferenceDate
        } else {
            return Date.timeIntervalSinceReferenceDate
        }
    }
}
